### PR TITLE
fix(tabs): Fix tab semantics and a11y

### PR DESCRIPTION
**Pull request by jsit**
Added a lot of semantic/a11y improvements to tabs.

Works for me locally but would be good to test yourself before approving.

https://getbootstrap.com/docs/5.3/components/navs-tabs/#javascript-behavior

### CODE REVIEW

File: src/shared/components/common/tabs.tsx [hunk 1]
```diff
@@ -10,44 +11,42 @@ interface TabsProps {
...
+          {this.props.tabs.map(({ key, label }, index) => (
             <li key={key} className="nav-item">
               <button
                 type="button"
-                className={`nav-link btn${
-                  this.state?.currentTab === key ? " active" : ""
-                }`}
-                onClick={linkEvent({ ctx: this, tab: key }, handleSwitchTab)}
+                data-bs-toggle="tab"
+                data-bs-target={`#${key}-tab-pane`}
```
Comment (dessalines) [inline src/shared/components/common/tabs.tsx:28]: Not going to work because we don't import bootstrap js.

Comment (dessalines) [reply src/shared/components/common/tabs.tsx:28]: I'd prefer that we don't import that, and instead use the existing inferno handlers for changing tabs. 

I'm all for using the bootstrap styling, but would prefer using our own javascript for event handling.

Comment (jsit) [reply src/shared/components/common/tabs.tsx:28]: Makes sense, thanks! Would still like to get that a11y markup in there so I’ll trim down this PR.

Comment (jsit) [reply src/shared/components/common/tabs.tsx:28]: Also in that case I made a recent change to the collapsing sidebar cards using Bootstrap JS that should be reverted then 😬

Comment (dessalines) [reply src/shared/components/common/tabs.tsx:28]: Thanks. Ya I agree that its pretty, but its a whole nother javascript dep to bring in. I think inferno has some animations now but its not worth it IMO.

**Review by SleeplessOne1917 — CHANGES_REQUESTED** [review]
Marking as request changes until the changes @dessalines requested are made.

File: src/shared/components/common/tabs.tsx (hunk 1 above)
Comment (jsit) [reply src/shared/components/common/tabs.tsx]: Yeah definitely understandable. I've removed the Bootstrap JS dependency but kept the a11y markup.

**Review by dessalines — APPROVED** [review]
I just tested and it works well on both. Might be nice if the background tabnames were muted / grayed, but that also might imply they're disabled. That can always be tweaked later.

Screenshot_2023-06-21-08-25-23-159_mark via gp

Thanks!

**Pull request merged**